arrows/vxl/register_algorithms.cxx

```diff
@@ -55,26 +54,127 @@ namespace kwiver {
 namespace arrows {
 namespace vxl {
 
-/// Register VXL algorithm implementations with the given or global registrar
-int register_algorithms( vital::registrar &reg )
+  extern "C"
+KWIVER_ALGO_VXL_EXPORT
+void
+register_factories( kwiver::vital::plugin_loader& vpm )
 {
-  REGISTRATION_INIT( reg );
-
-  REGISTER_TYPE( vxl::bundle_adjust );
-  REGISTER_TYPE( vxl::close_loops_homography_guided );
-  REGISTER_TYPE( vxl::estimate_canonical_transform );
-  REGISTER_TYPE( vxl::estimate_essential_matrix );
-  REGISTER_TYPE( vxl::estimate_fundamental_matrix );
-  REGISTER_TYPE( vxl::estimate_homography );
-  REGISTER_TYPE( vxl::estimate_similarity_transform );
-  REGISTER_TYPE( vxl::image_io );
-  REGISTER_TYPE( vxl::optimize_cameras );
-  REGISTER_TYPE( vxl::triangulate_landmarks );
-  REGISTER_TYPE( vxl::match_features_constrained );
-  REGISTER_TYPE( vxl::vidl_ffmpeg_video_input );
-
-  REGISTRATION_SUMMARY();
-  return REGISTRATION_FAILURES();
+  static auto const module_name = std::string( "algorithms.vxl" );
+  if (vpm.is_module_loaded( module_name ) )
+  {
+    return;
+  }
+
+  // add factory               implementation-name       type-to-create
+  auto fact = vpm.ADD_ALGORITHM( "vxl", kwiver::arrows::vxl::bundle_adjust );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_MODULE_NAME, module_name )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_VERSION, "1.0" )
+    .add_attribute( kwiver::vital::plugin_factory::PLUGIN_ORGANIZATION, "Kitware Inc." )
+    ;
+
+
+  fact = vpm.ADD_ALGORITHM( "vxl_homography_guided", kwiver::arrows::vxl::close_loops_homography_guided );
+  fact->add_attribute( kwiver::vital::plugin_factory::PLUGIN_DESCRIPTION,
+                    "" )
```

Comment:
Use this

    Use VXL to estimate a sequence of ground plane homographies to identify frames to match for loop closure